src/modules/cpu/kernel/noise_gaussian.hpp

```diff
@@ -0,0 +1,1442 @@
+#include "rppdefs.h"
+#include "rpp_cpu_simd.hpp"
+#include "rpp_cpu_common.hpp"
+
+RppStatus gaussian_noise_u8_u8_host_tensor(Rpp8u *srcPtr,
+                                           RpptDescPtr srcDescPtr,
+                                           Rpp8u *dstPtr,
+                                           RpptDescPtr dstDescPtr,
+                                           Rpp32f *meanTensor,
+                                           Rpp32f *stdDevTensor,
+                                           RpptXorwowStateBoxMuller *xorwowInitialStatePtr,
+                                           RpptROIPtr roiTensorPtrSrc,
+                                           RpptRoiType roiType,
+                                           RppLayoutParams layoutParams)
+{
+    RpptROI roiDefault = {0, 0, (Rpp32s)srcDescPtr->w, (Rpp32s)srcDescPtr->h};
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(dstDescPtr->n)
+    for(int batchCount = 0; batchCount < dstDescPtr->n; batchCount++)
+    {
+        RpptROI roi;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp32f mean = meanTensor[batchCount];
+        Rpp32f stdDev = stdDevTensor[batchCount];
+        Rpp32u offset = batchCount * srcDescPtr->strides.nStride;
+
+        Rpp8u *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        Rpp32u bufferLength = roi.xywhROI.roiWidth * layoutParams.bufferMultiplier;
+
+        Rpp8u *srcPtrChannel, *dstPtrChannel;
+        srcPtrChannel = srcPtrImage + (roi.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi.xywhROI.xy.x * layoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+
+        Rpp32u alignedLength = (bufferLength / 48) * 48;
+        Rpp32u vectorIncrement = 48;
+        Rpp32u vectorIncrementPerChannel = 16;
+
+        RpptXorwowStateBoxMuller xorwowState;
+        xorwowState.x[0] = xorwowInitialStatePtr[0].x[0] + offset;
```

Comment:
tbd:: keep this in vector of 8 and use AVX to update

Reply:
Ran a few experiments on this, turns out it doesn't make a difference if all the 5-6 params in the xorwow state are updated with the offset. We just need one param to be updated and that should suffice to achieve the desired randomization. So, currently updating just x[0] with offset.